astropy/coordinates/baseframe.py

```diff
@@ -945,6 +945,32 @@ def is_frame_attr_default(self, attrnm):
         """
         return attrnm in self._attr_names_with_defaults
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object is the same frame as the ``other`` object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes.  Note that it does *not* matter what, if any,
+        data either object has.
+
+        Parameters
+        ----------
+        other : BaseCoordinateFrame
+            the other frame to check
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if self.__class__ == other.__class__:
```

Comment:
My thought here was that it isn't obvious what the right thing is regarding the "don't care" issue, so I was refusing the temptation to guess by only allowing actual frames - the idea is that the user can always do ``SkyCoord(frame).is_equivalent_frame(othersc)`` instead of ``frame.is_equivalent_frame(othersc)`` if they really want this.  

Again, not a terribly strong opinion, though, so I could be convinced otherwise if there's a definite use case or others disagree.